juju/juju
# Remove juju- from k8s artifact names

## Description of change

When creating k8s artifacts like pods, volume claims, services etc, do not use the "juju-" prefix.
So that 2.5.0 deployed models continue to work, when an app is acted on, first see if it has been created using the legacy naming, and if so, continue to use the "juju-" prefixes for that app.

This started as just a change to k8s.go
When doing QA, I noticed that Juju doesn't handle well the case where the cluster is scaled down to 0 pods outside Juju. So add extra handling for that case.

## QA steps

bootstrap 2.5.0
deploy a k8s model with apps with storage
upgrade juju to 2.5.1
ensure that kubectl and juju status reflect a running model using the legacy names
scale out some apps and check
deploy some new apps with 2.5.1
ensure juju status shows a working system
scale out some apps and check
